packages/block-library/src/navigation/placeholder-preview.js

```diff
@@ -5,12 +5,12 @@ import { Icon, search } from '@wordpress/icons';
 
 const PlaceholderPreview = () => {
 	return (
-		<div className="wp-block-navigation-placeholder__preview">
-			<span className="wp-block-navigation-link"></span>
-			<span className="wp-block-navigation-link"></span>
-			<span className="wp-block-navigation-link"></span>
+		<ul className="wp-block-navigation-placeholder__preview wp-block-navigation__container">
+			<li className="wp-block-navigation-link">&#8203;</li>
```

Comment:
The markup and class changes here allow inheritance, and the `&#8203;` character is a zero width space which ensures the menu item has the correct height: to inherit line height the element has to have content inside.